### ISSUE TITLE
How to contribute?

### ISSUE THREAD

**Issue by ldecicco-USGS** [USGS-R/wateRuse#15]
Need to know about Rtools, roxygen, other packages to install.

ie initial steps to getting the R environment set up
- [x] install R
- [x] install RStudio
- [x] Rtools
- [x] install git

**Comment by dblodgett-usgs** [comment]
- [x] SA/PR required installs  
- [x] Project checkout  
- [x] RStudio setup  
- [x] Git setup
- [ ] Editing/committing

**Comment by dblodgett-usgs** [comment]
From @ldecicco-USGS 
The SA should install the latest versions of:
R: https://cran.r-project.org/bin/windows/base/
RStudio: https://www.rstudio.com/products/rstudio/download/
Rtools: https://cran.r-project.org/bin/windows/Rtools/
git: https://git-scm.com/download/win

**Comment by dblodgett-usgs** [comment]
Cloning Repository: (project checkout)
Fork into a personal repository:
https://help.github.com/articles/fork-a-repo/

Clone personal remote (origin) into local repository.
General: https://help.github.com/articles/cloning-a-repository/
RStudio: https://support.rstudio.com/hc/en-us/articles/200532077-Version-Control-with-Git-and-SVN

Use 'shell' function from RStudio git tab described in RStudio help above to do:
`git remote add upstream https://github.com/USGS-R/wateRuse.git`

See
`git config -l`
for more configurations that may be important such as user.name and user.email. May want to do:

```
git config --global user.email "mail@ewenharrison.com"
git config --global user.name "ewenharrison"
```

**Comment by dblodgett-usgs** [comment]
RStudio setup for development: https://support.rstudio.com/hc/en-us/articles/200486508-Building-Testing-and-Distributing-Packages

Generate documentation with roxygen need to be turned on.
https://support.rstudio.com/hc/en-us/articles/200486518-Customizing-Package-Build-Options

**Comment by dblodgett-usgs** [comment]
@jshourds-usgs I got a bunch of good links that address most of these. You will need to write up editing / committing.

**Comment by jshourds-usgs** [comment]
Okay so I am configuring my machine (while waiting for someone with PR access to update my R Studio version).  Oddly, I was able to download the rest of the required software.  Hopefully that is not a mistake.
My problem now is I seem to have added the master (git add remote upstream) but I can't seem to figure out how to add my local branch.  I seem to remember some git add origin but that didn't work.  
So here's what mine looks like:
![mygit](https://cloud.githubusercontent.com/assets/18489683/14993202/b0c8d724-1137-11e6-9efd-66d2c006bfc2.JPG)
 any ideas?

**Comment by dblodgett-usgs** [comment]
Looks like you pressed the wrong button. 😉  You actually wanted to do `git remote add upstream https://.....`

**Comment by dblodgett-usgs** [comment]
I think you somehow created multiple branches. Try...  
`git branch -v`
If that shows multiple entries, do:
`git checkout master`
then:
`bit branch -d upstream/master`

I think that should get you back to just having a local master.

**Comment by jshourds-usgs** [comment]
oh-- I did git add remote upstream. dyslexic much Jen?
I will try to clear myself and start over...

**Comment by jshourds-usgs** [comment]
I tried what you wrote 2 up and did okay until the last part
`bit branch -d upstream/master`
I got back bash: bit:  commant not found.
ugh

**Comment by dblodgett-usgs** [comment]
^^ bit should be git 🌟

**Comment by jshourds-usgs** [comment]
It's saying it's not found... I can see it!!!  R stands for Remedial for me on Tuesdays...

**Comment by dblodgett-usgs** [comment]
Then you should be fine. As long as you checked out master you are A - OK.

**Comment by jshourds-usgs** [comment]
:id: TEN T
So you think it's ok that I have a master and then an upstream/master?

**Comment by jshourds-usgs** [comment]
oh my now I have some detached head issue-- I do feel a bit decapitated...

**Comment by dblodgett-usgs** [comment]
To remove the remote tracking branch, it looks like you do
`git branch -d -r upstream/master`  
per: http://stackoverflow.com/questions/3046436/how-do-you-stop-tracking-a-remote-branch-in-git

**Comment by jshourds-usgs** [comment]
Glorious, that worked. Thank you!!!  :+1: 

So am I linked to my local branch?

**Comment by dblodgett-usgs** [comment]
The pull down in the screenshot above shows what branch you are on.

**Comment by jshourds-usgs** [comment]
Right-- I just have a master.
How do I add my branch?

**Comment by dblodgett-usgs** [comment]
As described above...

Look at `git config -l` Your remote origin should be your personal remote repository, your remote upstream should be the USGS-R repository. If either of them aren't there, use `git remote add ...` to add. or `git remote remove ...` then `git remote add ...` to straighten things out.

**Comment by jshourds-usgs** [comment]
so I have the master as the USGS-R repo.  and nothing else.  I tried `git remote add` and my personal remote repo link but it doesn't seem to be adding anything.  When I configure -l it only shows the USGS-R repo

**Comment by dblodgett-usgs** [comment]
You should probably start fresh. Setup a new project and use your personal fork: https://github.com/jshourds-usgs/wateRuse.git as the initial version control source when you go through the RStudio new project dialogue.

**Comment by jshourds-usgs** [comment]
Okay so I started fresh (writing some documentation along the way) and I am noticing a couple things that look different than they did in Wisc.  I seem to remember only an origin and a master in my git version list but I have this and I'm not sure what's what:  
![my versions](https://cloud.githubusercontent.com/assets/18489683/15025771/e30046a6-1208-11e6-95d5-61a3c8f0e016.JPG)
When I check out my `git config -l` here's what I get:
![myconfig](https://cloud.githubusercontent.com/assets/18489683/15025804/14474ce6-1209-11e6-90e2-2975976018ab.JPG)
Though my `git branch -v` looks like this:
![my git branch versions](https://cloud.githubusercontent.com/assets/18489683/15025854/5806cccc-1209-11e6-9b6c-f91d18386aa9.JPG)

Is all well here?  It seems pretty obvious which one master is in the git list but I don't know which is my origin-- origin/master? what is upstream/master? and what is origin/HEAD-> origin/master?

Thanks for any info

**Comment by dblodgett-usgs** [comment]
You are all set. I wouldn't worry about the origin/master and upstream/master... they are just the remote (origin and upstream) branches named the same as your local 'master' branch. branches are confusing enough, not to mention remote tracking branches.

What those options (origin and upstream master) should allow you to do is to switch the state of your code in RStudio to that of the remote repository without messing with your local 'master' branch. There are all kinds of tricky things that could be done with this, but we aren't going to use them.

**Comment by jshourds-usgs** [comment]
Okay thanks for all your help @dblodgett-usgs!  I think I am finally getting somewhere as I was able to load the time series data and see some plots-- :+1: :sunny:

**Comment by dblodgett-usgs** [comment]
Will be written up and closed by #38
